### PR TITLE
feat: add support for fastapi 0.111

### PR DESCRIPTION
**Pull request by fpletz**
[Contribution](README.md#contributing) checklist (recommended but not always applicable/required):

- [x] There's an _[automated test](tests/default.nix)_ for this change
- [ ] Commit messages or code include _references to related issues or PRs_ (including third parties)
- [x] Commit messages are _[conventional](https://www.conventionalcommits.org/)_ - examples from [the log](https://github.com/nix-community/poetry2nix/commits/master) include "feat: add changelog files to fixup hook", "fix(contourpy): allow wheel usage", and "test: add sqlalchemy2 test"

### CODE REVIEW

File: overrides/default.nix [hunk 1]
```diff
@@ -888,6 +888,13 @@ lib.composeManyExtensions [
         }
       );
 
+      fastapi-cli = prev.fastapi-cli.overridePythonAttrs (
+        old: lib.optionalAttrs (!(old.src.isWheel or false)) {
+          # remove the dependency cycle
+          propagatedBuildInputs = builtins.filter (p: p.pname != "fastapi") old.propagatedBuildInputs;
```
Comment (cpcloud) [inline overrides/default.nix:894]:
```suggestion
          propagatedBuildInputs = removePackagesByName (old.propagatedBuildInputs or []) [ self.fastapi ];
```

File: overrides/default.nix [hunk 2]
```diff
@@ -888,6 +888,13 @@ lib.composeManyExtensions [
         }
       );
 
+      fastapi-cli = prev.fastapi-cli.overridePythonAttrs (
+        old: lib.optionalAttrs (!(old.src.isWheel or false)) {
```
Comment (cpcloud) [inline overrides/default.nix:892]: This probably needs to be done regardless of whether `fastapi-cli` is a wheel, right?

Comment (fpletz) [reply overrides/default.nix:892]: Ah, yes, correct, I've copied that override block from another expression and didn't check this part again.

Comment (cpcloud) [reply overrides/default.nix]: Done

**Pull request merged**